compiler/rustc_typeck/src/check/regionck.rs

```diff
@@ -299,6 +299,29 @@ impl<'a, 'tcx> RegionCtxt<'a, 'tcx> {
         self.constrain_opaque_types(self.outlives_environment.free_region_map());
     }
 
+    fn visit_inline_const(&mut self, id: hir::HirId, body: &'tcx hir::Body<'tcx>) {
+        debug!("visit_inline_const(id={:?})", id);
+
+        // Save state of current function. We will restore afterwards.
+        let old_body_id = self.body_id;
+        let old_body_owner = self.body_owner;
+        let env_snapshot = self.outlives_environment.push_snapshot_pre_closure();
```

Comment:
same here, `push_snapshot_pre_closure` is somewhat badly named now

Reply:
Renamed to `push_snapshot_pre_typeck_child`. Feel free to suggest a better name :)